js/src/rive.ts

```diff
@@ -0,0 +1,965 @@
+  public load({src, buffer, autoplay = false}: RiveLoadParameters): void {
+    this._src = src;
+    this._buffer = buffer;
+    this._autoplay = autoplay;
+
+    // Stop all animations
+    self.stop();
+  
+    // If no source file url specified, it's a bust
+    if (!src && !buffer) {
+      console.error(Rive.missingErrorMessage);
+      return;
+    }
+  
+    // Reset internals
+    this._file = null;
+    this._artboard = null;
+    this._artboardName = null;
+    this._animations = [];
+    this._startingAnimationNames = [];
+    this._loaded = false;
+    
+
+
+    // Queue up play action and event if necessary
+    if (this._autoplay) {
+      this.taskQueue.add({ action: () => this.play() });
+    }
+  
+    // Wait for runtime to load
+    RuntimeLoader.awaitInstance().then((runtime) => {
+      this._rive = runtime;
+      // Load from a source uri or a data buffer
+      this.initialize();
+    });
+  }
+
+  // Sets a new layout
+  public set layout(layout: Layout) {
+    this._layout = layout;
+    if(!this.hasPlayingAnimations()) {
+      this.drawFrame();
+    }
+  }
+
+  // Returns the animation source, which may be undefined
+  public get source(): string {
+    return this._src;
+  }
+
+  // Returns a list of animation names on the chosen artboard
+  public get animationNames(): string[] {
+    // If the file's not loaded, we got nothing to return
+    if (!this._loaded) {
+      return [];
+    }
+    const animationNames: string[] = [];
+    for (let i = 0; i < this._artboard.animationCount(); i++) {
+      animationNames.push(this._artboard.animationAt(i).name);
+    }
+    return animationNames;
+  }
+
+  // Returns a list of playing animation names
+  public get playingAnimationNames(): string[] {
+    // If the file's not loaded, we got nothing to return
+    if (!this._loaded) {
+      return [];
+    }
+    return this._animations
+      .filter(a => !a.paused)
+      .map(a => a.name);
+  }
+
+  // Returns a list of paused animation names
+  public get pausedAnimationNames(): string[] {
+    // If the file's not loaded, we got nothing to return
+    if (!this._loaded) {
+      return [];
+    }
+    return this._animations
+      .filter(a => a.paused)
+      .map(a => a.name);
+  }
+
+  // Returns true if playing
+  public get isPlaying(): boolean {
+    return this._playback === playbackStates.play;
+  }
+
+  // Returns trus if all animations are paused
+  public get isPaused(): boolean {
+    return this._playback === playbackStates.pause;
+  }
+
+  // Returns true if all animations are stopped
+  public get isStopped(): boolean {
+    return this._playback === playbackStates.stop;
+  }
+
+  // Register a new listener
+  public on(type: EventType, callback: EventCallback) {
+    this.eventManager.add({
+      type: type,
+      callback: callback, 
+    });
+  }
+
+}
+
+
+// Loads Rive data from a URI via fetch.
+let loadRiveFile = async (src: string): Promise<ArrayBuffer> => {
```

Comment:
Same as above, I don't think you want to reassign this function. Using `const` or `function` might be better. Also no need to hard type return value if Typescript can infer it for you:
```typescript
// Typescript should be able to know from the code that the return value is Promise<ArrayBuffer>
function loadRiveFile(src: string) {
  const req = new Request(src);
  return fetch(req).then(res => res.arrayBuffer());
}
```

Reply:
I tend to write out the types to make it clear what's being returned. Let me make that const.